src/sensors/mod.rs

```diff
@@ -0,0 +1,97 @@
+//! Module defining common sensor functionality.
+
+mod humidity;
+mod people_now_present;
+mod power_consumption;
+mod temperature;
+
+pub use humidity::{HumiditySensor, HumiditySensorTemplate};
+pub use people_now_present::{PeopleNowPresentSensor, PeopleNowPresentSensorTemplate};
+pub use power_consumption::{PowerConsumptionSensor, PowerConsumptionSensorTemplate};
+pub use temperature::{TemperatureSensor, TemperatureSensorTemplate};
+
+use log::warn;
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+
+/// Common information describing any sensor.
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct SensorMetadata {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub location: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+}
+
+impl TryInto<LocalisedSensorMetadata> for SensorMetadata {
+    type Error = &'static str;
+
+    fn try_into(self) -> Result<LocalisedSensorMetadata, Self::Error> {
+        match self.location {
+            Some(location) => Ok(LocalisedSensorMetadata {
+                name: self.name,
+                location,
+                description: self.description,
+            }),
+            None => Err("No location specified when one is required"),
+        }
+    }
+}
```

Comment:
I'd want to get rid of this. Sensor templates should just have the correct SensorMetadata struct.